autoload/wintabs.vim

```diff
@@ -42,10 +42,41 @@ endfunction
 
 " close current tab
 function! wintabs#close()
-  call wintabs#refresh_buflist(0)
 
-  let size = len(w:wintabs_buflist)
   let buffer = bufnr('%')
+  let name = expand('%:t')
+  let msg = 'Do you want to save the changes you made to document '
+  let msg = msg . (name == '' ? "\"Untitled\"\n" : (name . "\n"))
+  let msg = msg . "Your changes will be lost if you don't save them."
```

Comment:
Tested this message in MacVim and terminal Vim (8). The message matches the GUI that MacVim shows.